### PR TITLE
Add warning to academy docs

### PR DESCRIPTION
**Pull request by andrewcoh**
### Useful links (Github issues, JIRA tickets, ML-Agents forum threads etc.)

https://github.com/Unity-Technologies/ml-agents/issues/4220

### Types of change(s)

- [ ] Bug fix
- [ ] New feature
- [ ] Code refactor
- [ ] Breaking change
- [ ] Documentation update
- [ ] Other (please describe)

### Checklist
- [ ] Added tests that prove my fix is effective or that my feature works
- [ ] Updated the [changelog](https://github.com/Unity-Technologies/ml-agents/blob/master/com.unity.ml-agents/CHANGELOG.md) (if applicable)
- [ ] Updated the [documentation](https://github.com/Unity-Technologies/ml-agents/tree/master/docs) (if applicable)
- [ ] Updated the [migration guide](https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Migrating.md) (if applicable)

### CODE REVIEW

File: docs/Learning-Environment-Design.md [hunk 1]
```diff
@@ -83,6 +83,12 @@ training a maze-solving agent, you would probably want to change the maze itself
 for each training episode. Otherwise, the agent would probably on learn to solve
 one, particular maze, not mazes in general.
 
+#### Academy Pitfalls
+
+In some games, it may be desirable to step the environment with the Academy manually via `Academy.Instance.EnvironmentStep()`.
```
Comment (vincentpierre) [inline docs/Learning-Environment-Design.md:88]: Strange that there is not a section about `Academy.Instance.EnvironmentStep()` already in this doc (out of scope for this PR). Stepping the Academy manually is an advanced scenario, maybe call this out?

File: docs/Learning-Environment-Design.md [hunk 2]
```diff
@@ -83,6 +83,12 @@ training a maze-solving agent, you would probably want to change the maze itself
...
+If this is the case, be sure `Academy.Instance.EnvironmentStep()` is not called directly or indirectly by the agent's
+`CollectObservations()` function as this will cause an infinite loop that prevents that main Academy update function from being called.
```
Comment (vincentpierre) [inline docs/Learning-Environment-Design.md:90]: CollectObservations is not the only method that might pose problems. Anything called BY the EnvironmentStep will be problematic. For example: Heuristic and OnActionReceived.

Comment (chriselion) [reply docs/Learning-Environment-Design.md]: Would it be better to guard against this? We could do something like
```
public void EnvironmentStep() {
  if(m_Stepping) {
    throw new UnityException("Infinite recursion prevented!");
  }
  m_Stepping = true;
  // do all the work
  m_Stepping = false;
}
```

Comment (andrewcoh) [reply docs/Learning-Environment-Design.md]: That works for me. It's up to you both.

Comment (chriselion) [reply docs/Learning-Environment-Design.md]: (with maybe a try/finally around the main body to reset the stepping flag, and/or change the throw to just return)

File: docs/Learning-Environment-Design.md [hunk 3]
```diff
@@ -83,6 +83,14 @@ training a maze-solving agent, you would probably want to change the maze itself
 for each training episode. Otherwise, the agent would probably on learn to solve
 one, particular maze, not mazes in general.
 
+#### Academy Pitfalls
+
+In some games, it may be desirable to step the environment with the Academy manually via `Academy.Instance.EnvironmentStep()`.
+Note, stepping manually is an advanced scenario and some care is required.
+If this is the case, be sure `Academy.Instance.EnvironmentStep()` is not called directly or indirectly by the agent's
+API e.g. from the `CollectObservations()`, `OnActionReceived`, and `Heuristic()` functions as this will
+cause an infinite loop that prevents the main Academy update function from being called.
```
Comment (vincentpierre) [inline docs/Learning-Environment-Design.md:92]:
```suggestion
cause an infinite loop that prevents the main Unity `Update()` function from being called.
```

File: docs/Learning-Environment-Design.md (hunk 2 above)
Comment (andrewcoh) [reply docs/Learning-Environment-Design.md]: @vincentpierre thoughts?

**Comment by andrewcoh** [comment]
Closing because this was made obsolete by #4227

**Pull request closed without merging**